store/LocalStore.js

```diff
@@ -9,17 +9,22 @@ function LocalStore (timeWindow, cache, app, continueExceeding) {
   this.continueExceeding = continueExceeding
 }
 
-LocalStore.prototype.incr = function (ip, cb) {
+LocalStore.prototype.incr = function (ip, cb, max) {
```

Comment:
adding the `max` here although it's not ideal because:
1. I can't pass it in the constructor as `max` can be a function that needs the request which we don't have/want access
2. passing it here and not before the `cb` as it would breaking change and we're tying to avoid it

Reply:
What that function would do?

Reply:
1. you don't need to move the `maximum` logic in index.js
2. you don't need to provide the `max` arg to the localstore file

Reply:
What would that function do I'm trying to understand, is it gonna check if `continueExceeding` is true and if so it would recalculate the TTL?

I'm thinking of a name for that function, maybe `recalculateTTLIfNeeded`? Or `recalculateTTL` and it wouldn't check for `continueExceeding` and we would check that in the place where we call that function

WDYT?

Reply:
The `index.js` file has the `current.coun` and the `max` values.

It would be something like

```
    if (current.count > max && theStore.forceDelete) {
      theStore.forceDelete(ip)
    }
```

Reply:
> The `index.js` file has the `current.coun` and the `max` values.
> 
> 
> 
> It would be something like
> 
> 
> 
> ```
> 
>     if (current.count > max && theStore.forceDelete) {
> 
>       theStore.forceDelete(ip)
> 
>     }
> 
> ```

@Eomm sorry for the late reply

This would cause a problem as the next increment would start from 0